installation-manager-core/src/main/java/com/codenvy/im/facade/InstallationManagerFacade.java

```diff
@@ -274,25 +230,22 @@ public UpdateArtifactInfo apply(Map.Entry<Artifact, Version> entry) {
 
         List<UpdateArtifactInfo> infos = new ArrayList<>();
 
-        infos.addAll(FluentIterable.from(allUpdates).transform(new Function<Map.Entry<Artifact, Version>, UpdateArtifactInfo>() {
-            @Override
-            public UpdateArtifactInfo apply(Map.Entry<Artifact, Version> entry) {
-                Artifact artifact = entry.getKey();
-                Version version = entry.getValue();
-
-                UpdateArtifactInfo info = UpdateArtifactInfo.createInstance(artifact.getName(), version.toString());
-                try {
-                    if (downloadManager.getDownloadedVersions(artifact).containsKey(version)) {
-                        info.setStatus(UpdateArtifactInfo.Status.DOWNLOADED);
-                    } else {
-                        info.setStatus(UpdateArtifactInfo.Status.AVAILABLE_TO_DOWNLOAD);
-                    }
-                } catch (IOException e) {
-                    // simply ignore
-                }
+        infos.addAll(FluentIterable.from(allUpdates).transform(entry -> {
```

Comment:
I think it is not hard to improve this code with stream API usage. Moreover you've already changed FluentIterable to Steam API in another place of this pull request https://github.com/codenvy/cdec/pull/26/files#diff-4c47e8f71dffba7574f409ab4b52444dL247
BTW Up to you